File: app/src/main/java/com/bearcave/passageplanning/waypoints/WaypointsManagerAdapter.java
package com.bearcave.passageplanning.waypoints;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.bearcave.passageplanning.R;
import com.bearcave.passageplanning.base.BaseManagerAdapter;
import com.bearcave.passageplanning.base.BaseManagerFragment;
import com.bearcave.passageplanning.waypoints.database.WaypointDAO;

import butterknife.ButterKnife;



public class WaypointsManagerAdapter extends BaseManagerAdapter<WaypointDAO, Integer>{


    public WaypointsManagerAdapter(BaseManagerFragment parent, Context context) {
        super(parent, context);
        addOption("Edit", element -> getDatabase().update(element));
        addOption("Delete", element -> {
            getDatabase().delete(element);
            getContainer().remove(element);
            notifyDataSetChanged();
        });

    }

    @Override
    public int getChildrenCount(int groupPosition) {
        return 1;
    }

    @Override
    public long getGroupId(int groupPosition) {
        return getContainer().get(groupPosition).getId();
    }

    @Override
    public long getChildId(int groupPosition, int childPosition) {
        return 0;
    }

    @Override
    public View getChildView(int groupPosition, int childPosition, boolean isLastChild, View convertView, ViewGroup parent) {
        View view = getInflater().inflate(R.layout.manager_item_details_item, parent, false);
        Waypoint waypoint = getContainer().get(groupPosition);

        TextView note = ButterKnife.findById(view, R.id.note);
        note.append(waypoint.getNote());

        TextView characteristic = ButterKnife.findById(view, R.id.characteristic);
        characteristic.append(waypoint.getCharacteristic());

        TextView gauge = ButterKnife.findById(view, R.id.gauge);
        gauge.append(waypoint.getGauge().getName());

        TextView ukc = ButterKnife.findById(view, R.id.ukc);
        ukc.append(String.valueOf(waypoint.getUkc()));

        TextView position = ButterKnife.findById(view, R.id.position);
        position.setText(String.valueOf(waypoint.getLongitude()) + "\n" + String.valueOf(waypoint.getLatitude()));

        return view;
    }

    @Override
    public boolean isChildSelectable(int groupPosition, int childPosition) {
        return false;
    }

}

File: app/src/main/java/com/bearcave/passageplanning/waypoints/WaypointsManagerAdapter.kt
package com.bearcave.passageplanning.waypoints

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import butterknife.ButterKnife
import com.bearcave.passageplanning.R
import com.bearcave.passageplanning.base.BaseManagerAdapter
import com.bearcave.passageplanning.waypoints.database.Waypoint


class WaypointsManagerAdapter(parent: WaypointsManagerFragment, context: Context) : BaseManagerAdapter<Waypoint, Int>(parent, context) {

    init {
        addOption(context.getString(R.string.action_edit), { dao -> parent.openEditor(dao)})
    }

    override fun getChildrenCount(groupPosition: Int): Int = 1

    override fun getGroupId(groupPosition: Int): Long = container[groupPosition].id.toLong()

    override fun getChildId(groupPosition: Int, childPosition: Int): Long = 0

    override fun getChildView(groupPosition: Int, childPosition: Int, isLastChild: Boolean, convertView: View?, parent: ViewGroup?): View {
        val view = convertView ?: inflater.inflate(R.layout.waypoint_child_item, parent, false)

        val wpt = container[groupPosition]

        // note
        ButterKnife.findById<TextView>(view!!, R.id.note)
            .text = "Note ${wpt.note}"

        // characteristic
        ButterKnife.findById<TextView>(view, R.id.characteristic)
            .text = "Characteristic: ${wpt.characteristic}"

        // gauge
        ButterKnife.findById<TextView>(view, R.id.gauge)
            .text = "Gauge: ${wpt.gauge.humanCode}"

        // ukc
        ButterKnife.findById<TextView>(view, R.id.ukc)
            .text = "${context.getString(R.string.ukc)} ${wpt.ukc}"

        // latitude
        ButterKnife.findById<TextView>(view, R.id.latitude)
            .text = wpt.latitudeInSecondFormat

        // longitude
        ButterKnife.findById<TextView>(view, R.id.longitude)
                .text = wpt.longitudeInSecondFormat

        return view
    }

    override fun isChildSelectable(groupPosition: Int, childPosition: Int): Boolean = false
}

